### PR TITLE
Add '--fork=' to EvmTool

### PR DESCRIPTION
**Pull request by shemnon**
## PR description

Add a CLI '--fork' option to allow code to be evaluated in specific forks.

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

**Comment by shemnon** [comment]
docs change

a new `--fork=` option added to the EVM tool (if that tool is documented)

Samples:

```
$ ./ethereum/evmtool/build/install/evmtool/bin/evm --fork FutureEips --code "0xEF0001 01000C 020003 000b 0002 0008 030000 00 00000002 02010002 01000002 60016002b00001b00002b1 01b1 60005360106000f3" --json
```

```
$ ./ethereum/evmtool/build/install/evmtool/bin/evm --code "0xEF0001 01000C 020003 000b 0002 0008 030000 00 00000002 02010002 01000002 60016002b00001b00002b1 01b1 60005360106000f3" --json
```

File: ethereum/evmtool/src/main/java/org/hyperledger/besu/evmtool/MainnetGenesisFileModule.java [hunk 1]
```diff
@@ -31,14 +41,96 @@ class MainnetGenesisFileModule extends GenesisFileModule {
...
+        return schedule.get();
+      }
+    }
     return MainnetProtocolSchedule.fromConfig(configOptions, EvmConfiguration.DEFAULT);
   }
 
-  @Override
-  BlockHeaderFunctions blockHashFunction() {
-    return new MainnetBlockHeaderFunctions();
+  public static Map<String, Supplier<HeaderBasedProtocolSchedule>> createSchedules() {
```
Comment (garyschulte) [inline ethereum/evmtool/src/main/java/org/hyperledger/besu/evmtool/MainnetGenesisFileModule.java:63]: This looks like a new place to remember to update when we create new forks.  I wish we had a way of discovering these places without resorting to some kind of rigid static analysis/annotation

**Review by garyschulte — APPROVED** [review]

File: ethereum/evmtool/src/main/java/org/hyperledger/besu/evmtool/MainnetGenesisFileModule.java (hunk 1 above)
Comment (shemnon) [reply ethereum/evmtool/src/main/java/org/hyperledger/besu/evmtool/MainnetGenesisFileModule.java:63]: The main issue here is that it crosses module barriers.  EVM has configurations that need to be set up, and so does the mainnet client.  Throw in if we do things like separate out the block and tx data validation that those also have different rules based on different forks.

But if we had all EVM changes come from one source that would be good.  Will think about it once EOF settles down.

**Comment by shemnon** [comment]
One big change I want to do with EVM tool is remove the ability to interface with a production database and move those features into another tool that can also do things like extract contract addresses and storage values.

**Comment by GesusBitcoin** [comment]
Available as a browser extension and as a mobile app, MetaMask equips you with a key vault, secure login, token wallet, and token exchange—everything you need to manage your digital assets.

**Pull request merged**